### DISCUSSION TITLE
HSL colours with the ring utility

### DISCUSSION THREAD

**Discussion by Dharmatologist** [tailwindlabs/tailwindcss#13407] [Help]
Hi guys,

Has anyone encountered this same issue? When I use the `ring` utility (specifically, `focus:ring-<color>` and `focus-within:ring-<color>` but replicable on a basic `ring--<color>` too), where `color` is defined in my tailwind config as `hsl(h,s,l)` the output is as if the ring is transparent, i.e. not visible.

I can work around it by creating a few hex versions of colours but ideally all utilities work with HSL colours...

For example:
```
<div class="ring-mycolor-500 ring-2"></div>
```

Tailwind.config.js
```
(...)
colors: {
  mycolor {
        500: "hsl(259, 25, 41)",
  },
},
(...)
```

**Comment by alex-krasikau** [comment]
Hey @Dharmatologist!

Your example seems to work fine for me on [Play Tailwind CSS - HSL ring color](https://play.tailwindcss.com/Ln2Hf4vxXy).

I suggest going through this example and ensuring that you have the correct [Tailwind CSS setup](https://dev.to/plainsailing/tailwind-css-not-working-its-probably-this-9m4).

If it still doesn't work, could you create a replication example on Play Tailwind or provide some screenshots of the devtools that show your current issue?

**Comment by Dharmatologist** [comment]
Hi!
Thanks for reaching out.

The sample did not work for me either, and I fixed it by adding the % signs
for S and L in the 'mycolor' definition.

I then did the same in my project, and boom! it works. So it was an issue
with my color definitions, which lacked the % token in for example
"hsl(259, 10%, 10%)".

I don't know why the sample worked for you and not for me (on my phone as
well as my laptop) but in the end, the % token is technically required IIRC
so it was all my bad!

(Tailwind seems happy to ignore the omission when using my colors in other
utils such as bg-, text-, etc and only the ring- utilities were affected)

Thanks for helping me get to the bottom of this!

Ed
